test/stdcpp/Makefile

```diff
@@ -6,23 +6,22 @@ TESTS:=allocator new
 TESTS11:=array
 TESTS17:=string_view
 OLDABITESTS:=
-LIBCPPTESTS:=
 
 ifeq (osx,$(OS))
-	TESTS+=string
+#	TESTS+=string
```

Comment:
@Geod24 @jacob-carlborg
Can one of you guys uncomment this line and test on your OSX environments?
There's an error running the unit tests which I have no way to resolve. I'm fairly confident I have the struct layout correct for Apple, so maybe there's a calling convention thing?

Reply:
Fucking typical that Apple can't just use libc++ unmolested! They just have to change some shit don't they!!

Reply:
Okay, looking at it now. The `size` of "Hello" is 36 on the C++ size, and the iteration starts on `e`, not `H`. Will see if I can spot what's wrong.

Reply:
Check that `_LIBCPP_ABI_ALTERNATE_STRING_LAYOUT` is appropriate for OSX?
I'm pretty sure it is... Evidence I could find suggested osx rocks the alternate ABI...

Reply:
Check 1, does the by-ref pass correctly? If not, the structure is wrong. If it does, and by-val doesn't work, it's probably calling convention.

Reply:
No, `_LIBCPP_ABI_ALTERNATE_STRING_LAYOUT` does not seem to be enabled by default.
Disabling it make the first 2 lines pass, but then after assignment you'll get:
```
string(39567,0x109c275c0) malloc: *** error for object 0x7fdc29c02d10: pointer being freed was not allocated
string(39567,0x109c275c0) malloc: *** set a breakpoint in malloc_error_break to debug
```

Reply:
If the struct looks right, and the values carry correctly from D to C++, maybe check that allocator.d is doing the right thing?
Maybe the allocation is being aligned or something? Windows does that, but I didn't see that business in libc++. Maybe Apple do something special in the allocator?
(I would expect the allocator tests to fail if they do)

Reply:
@TurkeyMan if you need access to a Mac you can use Circle CI. You can setup a job that fails immediately. For failed jobs you get the option, in the UI, to rerun the job with SSH enabled. Then you will have a session available for one hour.

Reply:
I mean, that's very helpful information, but my personal policy is not to engage with the devil ;)

Reply:
Well, your loss 😃.

Reply:
Not really. I'll never be upset that the Mac doesn't work :P